src/components/documentManager.ts

```diff
@@ -0,0 +1,678 @@
+import { parseISO8601Date } from '../helpers';
+import { JellyfinApi } from './jellyfinApi';
+import { deviceIds, getActiveDeviceId } from './castDevices';
+import { BaseItemDto } from '~/api/generated/models/base-item-dto';
+
+export abstract class DocumentManager {
+    // Duration between each backdrop switch in ms
+    private static backdropPeriodMs: number | null = 30000;
+    // Timer state - so that we don't start the interval more than necessary
+    private static backdropTimer = 0;
+
+    // TODO make enum
+    private static status = '';
+
+    /**
+     * Hide the document body on chromecast audio to save resources
+     */
+    public static initialize(): void {
+        if (getActiveDeviceId() === deviceIds.AUDIO)
+            document.body.style.display = 'none';
+    }
+
+    /**
+     * Set the background image for a html element, with image preloading.
+     *
+     * @param {HTMLElement} element HTML Element
+     * @param {string | null} src URL to the image or null to remove the active one
+     * @returns {Promise<void>} wait for the background to be switched
+     */
+    private static setBackgroundImage(
+        element: HTMLElement,
+        src: string | null
+    ): Promise<void> {
+        if (src) {
+            return new Promise((resolve, reject) => {
+                const preload = new Image();
+                preload.src = src;
+                // TODO event listener for reject / load failure
+                preload.addEventListener('load', () => {
+                    requestAnimationFrame(() => {
+                        element.style.backgroundImage = `url(${src})`;
+                        resolve();
+                    });
+                });
+                preload.addEventListener('error', () => {
+                    reject();
+                });
+            });
+        } else {
+            element.style.backgroundImage = '';
+            return Promise.resolve();
+        }
+    }
+
+    /**
+     * Get url for primary image for a given item
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {string | null} url to primary image
+     */
+    private static getPrimaryImageUrl(item: BaseItemDto): string | null {
+        if (item.AlbumPrimaryImageTag)
+            return JellyfinApi.createUrl(
+                `Items/${item.AlbumId}/Images/Primary?tag=${item.AlbumPrimaryImageTag}`
+            );
+        else if (item.ImageTags?.Primary)
+            return JellyfinApi.createUrl(
+                `Items/${item.Id}/Images/Primary?tag=${item.ImageTags.Primary}`
+            );
+        else return null;
```

Comment:
We should probably add a rule for this, but imo all the blocks should have braces. **Don't fix it now, we can do it automatically with eslint in another PR**.